### PR TITLE
Relax the test assertion and avoid sleep usage

### PR DESCRIPTION
**Pull request by lacasseio**
### Context
Related this [CI failure](https://builds.gradle.org/viewLog.html?buildId=3123404&buildTypeId=Gradle_Release_CommitPhase_LinuxCommit_LinuxCommitJava18_7LinuxCommitJava18) and this [Slack discussion](https://gradle.slack.com/archives/C03A0MT6S/p1496670232626754).

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Branches_Checkpoints&branch_Gradle_Branches_Checkpoints=dl-fix-flaky-console-test)
- [ ] Verify documentation including proper use of `@since` and `@Incubating` annotations for all public APIs
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by lacasseio** [comment]
CI Stage 3 pass.

File: subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/BasicGroupedTaskLoggingFunctionalSpec.groovy [hunk 1]
```diff
@@ -18,11 +18,18 @@ package org.gradle.internal.logging
...
 import org.gradle.internal.logging.sink.GroupingProgressLogEventGenerator
+import org.gradle.test.fixtures.ConcurrentTestUtil
+import org.gradle.test.fixtures.server.http.CyclicBarrierHttpServer
+import org.junit.Rule
 import spock.lang.IgnoreIf
 import spock.lang.Issue
 
 class BasicGroupedTaskLoggingFunctionalSpec extends AbstractConsoleFunctionalSpec {
+    @Rule
+    CyclicBarrierHttpServer server = new CyclicBarrierHttpServer()
```
Comment (big-guy) [inline subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/BasicGroupedTaskLoggingFunctionalSpec.groovy]: We're trying to phase this out, so we should use `BlockingHttpServer`

Comment (lacasseio) [inline subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/BasicGroupedTaskLoggingFunctionalSpec.groovy]: Done.

**Comment by bmuschko** [comment]
@lacasseio Is this ready to be merged?

**Pull request merged**